### PR TITLE
New Resource: `azurerm_dynatrace_monitor`

### PR DESCRIPTION
**Pull request by jiaweitao001**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* New Resource: `azurerm_dynatrace_monitor`

This is a (please select all that apply):

- [ ] Bug Fix
- [x] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

As discussed several months ago, since the tests of this resource cost too much, I'll perform tests on the subs provided by the service team. Here's the test results.

```
PASS: TestAccDynatraceMonitor_basic (394.27s)
PASS: TestAccDynatraceMonitor_update(456.12s)
PASS: TestAccDynatraceMonitor_requiresImport(387.24s)
```

### CODE REVIEW

File: internal/services/dynatrace/helper.go
Comment (katbyte) [inline internal/services/dynatrace/helper.go:178]: could we please just inline these?
```suggestion
	return []UserInfo{
		{
			Country:      v["country"].(string),
			EmailAddress: v["email"].(string),
			FirstName:    v["first_name"].(string),
			LastName:     lastName,
			PhoneNumber:  phoneNumber,
		},
	}
```

etc

File: internal/services/dynatrace/helper.go
Comment (katbyte) [inline internal/services/dynatrace/helper.go:23]: Shouldn't these be an enum?

File: internal/services/dynatrace/helper.go
Comment (katbyte) [inline internal/services/dynatrace/helper.go:38]: and this

File: internal/services/dynatrace/helper.go
Comment (katbyte) [inline internal/services/dynatrace/helper.go:68]: we can propertly validate this is an email?

File: internal/services/dynatrace/dynatrace_monitors_resource.go
Comment (katbyte) [inline internal/services/dynatrace/dynatrace_monitors_resource.go:84]: is there a reason we are pulling these out into a function? they are only used here so could we please inline them to be consistant with the rest of the provider?

File: internal/services/dynatrace/dynatrace_monitors_resource.go
Comment (katbyte) [inline internal/services/dynatrace/dynatrace_monitors_resource.go:76]: shouldn't this be an enum?

File: website/docs/r/dynatrace_monitors.html.markdown
Comment (katbyte) [inline website/docs/r/dynatrace_monitors.html.markdown:82]: comma's are not used for two item list
```suggestion
* `billing_cycle` - (Optional) Different billing cycles. Possible values are `MONTHLY` and `WEEKLY`.
```

File: website/docs/r/dynatrace_monitors.html.markdown
Comment (katbyte) [inline website/docs/r/dynatrace_monitors.html.markdown:62]: and here

File: website/docs/r/dynatrace_monitors.html.markdown
Comment (katbyte) [inline website/docs/r/dynatrace_monitors.html.markdown:88]: and here

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/dynatrace/dynatrace_monitors_resource.go
Comment (jiaweitao001) [reply internal/services/dynatrace/dynatrace_monitors_resource.go:76]: Will change.

File: internal/services/dynatrace/dynatrace_monitors_resource.go
Comment (jiaweitao001) [reply internal/services/dynatrace/dynatrace_monitors_resource.go:84]: Will inline them.

File: internal/services/dynatrace/helper.go
Comment (jiaweitao001) [reply internal/services/dynatrace/helper.go:23]: They are not enums.

File: internal/services/dynatrace/helper.go
Comment (jiaweitao001) [reply internal/services/dynatrace/helper.go:38]: Same as above, these are not enums.

File: internal/services/dynatrace/helper.go
Comment (jiaweitao001) [reply internal/services/dynatrace/helper.go:68]: Will do.

File: internal/services/dynatrace/helper.go
Comment (jiaweitao001) [reply internal/services/dynatrace/helper.go:178]: Will do.

File: website/docs/r/dynatrace_monitors.html.markdown
Comment (jiaweitao001) [reply website/docs/r/dynatrace_monitors.html.markdown:62]: Will change.

File: website/docs/r/dynatrace_monitors.html.markdown
Comment (jiaweitao001) [reply website/docs/r/dynatrace_monitors.html.markdown:82]: Will change.

File: website/docs/r/dynatrace_monitors.html.markdown
Comment (jiaweitao001) [reply website/docs/r/dynatrace_monitors.html.markdown:88]: Will change.

File: internal/services/dynatrace/helper.go
Comment (katbyte) [reply internal/services/dynatrace/helper.go]: Other APIs correctly make these an enum (ie sql_license_type) so i presume this is a swagger oversite, could we create a quick rest api specs issue and the link that inline?

**Review by katbyte — APPROVED** [review]
3 minor comments + the rest API specs for the missing enum but otherwise LGTM once they are resolved 🏷️

**Comment by katbyte** [comment]
@jiaweitao001 also don't forget to post test evidence

**Comment by jiaweitao001** [comment]
> @jiaweitao001 also don't forget to post test evidence

Hi @katbyte , the test results are listed in the description note at the top.

**Comment by katbyte** [comment]
@jiaweitao001 - you need to rerun the tests with all the changes/commits made to ensure it still works, and tests should be a comment after the changes have been made. Please re-run the tests and add as a comment.

**Comment by jiaweitao001** [comment]
> @jiaweitao001 - you need to rerun the tests with all the changes/commits made to ensure it still works, and tests should be a comment after the changes have been made. Please re-run the tests and add as a comment.

Latest test results:

```
=== RUN   TestAccDynatraceMonitor_basic
=== PAUSE TestAccDynatraceMonitor_basic
=== RUN   TestAccDynatraceMonitor_update
=== PAUSE TestAccDynatraceMonitor_update
=== RUN   TestAccDynatraceMonitor_requiresImport
=== PAUSE TestAccDynatraceMonitor_requiresImport
=== CONT  TestAccDynatraceMonitor_basic
=== CONT  TestAccDynatraceMonitor_requiresImport
=== CONT  TestAccDynatraceMonitor_update
--- PASS: TestAccDynatraceMonitor_requiresImport (293.30s)
--- PASS: TestAccDynatraceMonitor_basic (300.87s)
--- PASS: TestAccDynatraceMonitor_update (328.71s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/dynatrace     328.748s
```

**Review by katbyte — APPROVED** [review]
Thanks LGTM ☕

**Pull request merged**
